File: app/src/main/java/solutions/alterego/android/unisannio/ingegneria/IngegneriaAvvisiDipartimentoParser.java
package solutions.alterego.android.unisannio.ingegneria;

import android.support.annotation.NonNull;
import java.util.UUID;
import org.joda.time.DateTime;

import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.safety.Whitelist;
import org.jsoup.select.Elements;

import java.util.ArrayList;
import java.util.List;

import solutions.alterego.android.unisannio.interfaces.Parser;
import solutions.alterego.android.unisannio.models.Article;
import solutions.alterego.android.unisannio.utils.DateUtils;

public class IngegneriaAvvisiDipartimentoParser implements Parser<Article> {

    @NonNull @Override
    public List<Article> parse(@NonNull Document document) {
        List<Article> list = new ArrayList<>();

        Elements elements = document.select("item");


         for (Element element: elements){
             String title = element.select("title").text();
             String body = clearHtml(element.select("description").text());
             String date = element.select("pubdate").text();
             String author = element.select("author").text();
             String link = element.select("guid").text();

             DateTimeFormatter dtf = DateTimeFormat.forPattern("dd/MM/yyyy");
             DateTime jodatime = dtf.parseDateTime(DateUtils.convertMonth(date.substring(5)));

             list.add(new Article(UUID.randomUUID().toString(), title, "", link, body, jodatime.toString()));
         }

        return list;
    }

    //Method that remove all the htlm tag from the body of the article.
    private String clearHtml(String html){
        String tmp1 = Jsoup.clean(html,Whitelist.none());
        return tmp1.replace("&nbsp;","").replace("&gt;","");
    }
}

File: app/src/main/java/solutions/alterego/android/unisannio/ingegneria/IngegneriaAvvisiDipartimentoParser.kt
package solutions.alterego.android.unisannio.ingegneria

import org.joda.time.format.DateTimeFormat
import org.jsoup.Jsoup
import org.jsoup.nodes.Document
import org.jsoup.safety.Whitelist
import solutions.alterego.android.unisannio.interfaces.Parser
import solutions.alterego.android.unisannio.models.Article
import solutions.alterego.android.unisannio.utils.DateUtils
import java.util.*

class IngegneriaAvvisiDipartimentoParser : Parser<Article> {


    override fun parse(document: Document): List<Article> {

        val list = ArrayList<Article>()

        val elements = document.select("item")


        for (element in elements) {
            val title = element.select("title").text()
            val body = clearHtml(element.select("description").text())
            val date = element.select("pubdate").text()
            val author = element.select("author").text()
            val link = element.select("guid").text()

            val dtf = DateTimeFormat.forPattern("dd/MM/yyyy")
            val jodatime = dtf.parseDateTime(DateUtils.convertMonth(date.substring(5)))

            list.add(Article(UUID.randomUUID().toString(), title, author, link, body, jodatime.toString()))
        }

        return list
    }

    private fun  clearHtml(html: String?): String {

        val tmp1 = Jsoup.clean(html, Whitelist.none())
        val removed_tag = tmp1.replace("&nbsp;", "").replace("&gt;", "")

        return removed_tag
    }
}